### ISSUE TITLE
VOS bleadperl test failure on 16539

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#5449]
Migrated from [rt.perl.org#9279](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=9279) (status was 'resolved')

Searchable as RT9279$

**Comment by p5pRT** [comment]
### From @paulg1973

#### Created by @paulg1973



The new ext/B/t/o\.t test case fails on VOS\. We seem to be returning the
output lines in a different order than the test expects\.  I imagine this
output is probably coming from several children via dup and fork\.  I tried
to find "runperl" so I could see what it is doing\, but I can't find it in
perl/\*\.c\, perl/\*\.h or pod/\*\.pod\, O\.pm or B\.pm or any of the modules they
call\. Where is perl hiding this thing?

The output of the test is below\. I hacked it up to show the lines of output
it is processing\.

Thanks
PG


bash\-2\.04$ perl o\.t
1\.\.9
ok 1 \- require O
\# lines\[0\] = Compiling\!
\# lines\[1\] = \(foo\) \<bar>
\# lines\[2\] = \[\]
\# lines\[3\] = \-e syntax OK
not ok 2 \- O\.pm should not munge perl output without \-qq
\# Failed at o\.t line 38
\#      got 'Compiling\!'
\# expected '\-e syntax OK'
not ok 3 \- Output should not be saved without \-q switch
\# Failed at o\.t line 39
\#      got '\(foo\) \<bar>'
\# expected 'Compiling\!'
not ok 4 \- O\.pm should call backend compile\(\) method
\# Failed at o\.t line 40
\#      got '\[\]'
\# expected '\(foo\) \<bar>'
not ok 5 \- Nothing should be in $O&#8203;::BEGIN\_output without \-q
\# Failed at o\.t line 41
\#      got '\-e syntax OK'
\# expected '\[\]'
ok 6 \- Output should not be printed with \-q switch
ok 7 \- \.\.\. but should be in $O&#8203;::BEGIN\_output
ok 8 \- \-qq should suppress even the syntax OK message
not ok 9 \- O\.pm should die if backend compile\(\) does not return a subref
\# Failed at o\.t line 59
\#      got 'Compiling\!'
\# expected /\(?\-xism&#8203;:fail at \.eval\)/
bash\-2\.04$

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=high

Site configuration information for perl v5.7.3:

Configured by Green at Fri May 10 18:39:24 edt 2002.

Summary of my perl5 (revision 5.0 version 7 subversion 3 patch 16538)
configuration:
  Platform:
    osname=vos, osvers=vos, archname=hppa1.1
    uname='vos m30 vos release 14.5.beta.af vos release 14.5.beta.af
continuum 629 unknown '
    config_args='-Dusedevel -de -DDEBUGGING -Doptimize=-g'
    hint=recommended, useposix=true, d_sigaction=define
    usethreads=undef use5005threads=undef useithreads=undef
usemultiplicity=undef
    useperlio=define d_sfio=undef uselargefiles=undef usesocks=undef
    use64bitint=undef use64bitall=undef uselongdouble=undef
    usemymalloc=n, bincompat5005=undef
  Compiler:
    cc='gcc', ccflags ='-D_SVID_SOURCE -D_POSIX_C_SOURCE=199509L -DDEBUGGING
-fno-strict-aliasing -I/system/stcp/include_library
-I/system/include_library/sysv',
    optimize='-g',
    cppflags='-D_SVID_SOURCE -D_POSIX_C_SOURCE=199509L -DDEBUGGING
-fno-strict-aliasing -I/system/stcp/include_library
-I/system/include_library/sysv'
    ccversion='', gccversion='2.95.2 19991024 (release)', gccosandvers=''
    intsize=4, longsize=4, ptrsize=4, doublesize=8, byteorder=4321
    d_longlong=define, longlongsize=8, d_longdbl=define, longdblsize=8
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t',
lseeksize=4
    alignbytes=8, prototype=define
  Linker and Libraries:
    ld='ld', ldflags =' -L/system/stcp/object_library
-L/system/stcp/object_library/common -L/system/stcp/object_library/net
-L/system/stcp/object_library/socket -L/system/posix_object_library/sysv
-L/system/posix_object_library -L/system/c_object_library
-L/system/object_library'
    libpth=/system/stcp/object_library /system/stcp/object_library/common
/system/stcp/object_library/net /system/stcp/object_library/socket
/system/posix_object_library/sysv /system/posix_object_library
/system/c_object_library /system/object_library
    libs=
    perllibs=
    libc=, so=none, useshrplib=false, libperl=libperl.a
  Dynamic Linking:
    dlsrc=dl_none.xs, dlext=none, d_dlsymun=undef, ccdlflags=''
    cccdlflags='', lddlflags=''

Locally applied patches:
    DEVEL16538


@INC for perl v5.7.3:
    lib
    /system/ported/perl5/lib/5.7.3/hppa1.1
    /system/ported/perl5/lib/5.7.3
    /system/ported/perl5/lib/site_perl/5.7.3/hppa1.1
    /system/ported/perl5/lib/site_perl/5.7.3
    /system/ported/perl5/lib/site_perl
    .


Environment for perl v5.7.3:
    HOME=/%es#lang/Languages/Green
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
 
PATH=.:/%es#lang/Languages/Green/bin:/system/gnu_library/bin:/bin:/%es#lang/
Languages/Green/library:/%es#lang/Languages/Green/library.hppa:/system/es_li
brary:/system/scs/command_library:/system/ported/command_library:/system/itl
_command_library:/system/command_library:/system/tools_library:/system/qts_c
ommand_library:/system/css_command_library:/system/maint_library:/system/app
lications_library:/system/es_library/obsolete:/system/stcp/command_library:/
system/MIS/mis_library
    PERL_BADLANG (unset)
    SHELL=/system/gnu_library/bin/bash.pm





```
</details>

**Comment by p5pRT** [comment]
### From @jhi

On Sat\, May 11\, 2002 at 08&#8203;:17&#8203;:12PM \-0400\, Green\, Paul wrote&#8203;:
> This is a bug report for perl from Paul\.Green@&#8203;stratus\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.7\.3\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> The new ext/B/t/o\.t test case fails on VOS\. We seem to be returning the
> output lines in a different order than the test expects\.  I imagine this
> output is probably coming from several children via dup and fork\.  I tried
> to find "runperl" so I could see what it is doing\, but I can't find it in
> perl/\*\.c\, perl/\*\.h or pod/\*\.pod\, O\.pm or B\.pm or any of the modules they
> call\. Where is perl hiding this thing?

t/test\.pl

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
### From @jhi

On Sat\, May 11\, 2002 at 08&#8203;:17&#8203;:12PM \-0400\, Green\, Paul wrote&#8203;:
> This is a bug report for perl from Paul\.Green@&#8203;stratus\.com\,
> generated with the help of perlbug 1\.33 running under perl v5\.7\.3\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> \[Please enter your report here\]
> 
> The new ext/B/t/o\.t test case fails on VOS\. We seem to be returning the
> output lines in a different order than the test expects\.  I imagine this
> output is probably coming from several children via dup and fork\.  I tried
> to find "runperl" so I could see what it is doing\, but I can't find it in
> perl/\*\.c\, perl/\*\.h or pod/\*\.pod\, O\.pm or B\.pm or any of the modules they
> call\. Where is perl hiding this thing?

t/test\.pl

\-\- 
$jhi\+\+; \# http&#8203;://www\.iki\.fi/jhi/
&nbsp;        \# There is this special biologist word we use for 'stable'\.
&nbsp;        \# It is 'dead'\. \-\- Jack Cohen

**Comment by p5pRT** [comment]
@paulg1973 - Status changed from 'open' to 'resolved'

**Comment by p5pRT** [comment]
### From @paulg1973

This problem was fixed by change \#16554\.

Thanks
PG
